### PR TITLE
[Merged by Bors] - Rename node_state.sql to local.sql

### PR DESCRIPTION
**Pull request by fasmat**
## Motivation

`node_state.sql` seems to be confused with the node's `state.sql` regularly. To avoid confusion, rename it to `local.sql`

## Description

- Add code that renames `node_state.sql` to `local.sql`
  - uses `VACUUM INTO` to create a new file containing all content from `node_state.sql`
  - renames `node_state.sql` to `node_state.sql.bak`
  - fails when both DB exist
  - no-op if `node_state.sql` doesn't exist
- Update CHANGELOG to use the new name instead of the old

## Test Plan

Added tests for migration code

## TODO

- [x] Explain motivation or link existing issue(s)
- [x] Test changes and document test plan
- [x] Update documentation as needed
- [x] Update [changelog](../CHANGELOG.md) as needed

### CODE REVIEW

File: node/node.go [hunk 1]
```diff
@@ -1775,6 +1784,64 @@ func (app *App) setupDBs(ctx context.Context, lg log.Log) error {
...
+		sql.WithConnections(app.Config.DatabaseConnections),
+	)
+	if err != nil {
+		return fmt.Errorf("open sqlite db %w", err)
+	}
+	defer oldDB.Close()
+
+	if _, err := oldDB.Exec(fmt.Sprintf("VACUUM INTO '%s'", dbFile), nil, nil); err != nil {
+		return fmt.Errorf("vacuum %s to %s: %w", oldDBFile, dbFile, err)
+	}
```
Comment (poszu) [inline node/node.go:1838]: Would it make sense to use https://pkg.go.dev/github.com/go-llsqlite/llsqlite#Conn.BackupToDB?

File: node/node.go [hunk 2]
```diff
@@ -1775,6 +1784,64 @@ func (app *App) setupDBs(ctx context.Context, lg log.Log) error {
...
+	if err := atomic.ReplaceFile(oldDBFile, fmt.Sprintf("%s.bak", oldDBFile)); err != nil {
+		return fmt.Errorf("renaming %s to %s: %w", oldDBFile, fmt.Sprintf("%s.bak", oldDBFile), err)
+	}
```
Comment (poszu) [inline node/node.go:1844]: What about the other files (-wal, -shm)?

Comment (fasmat) [reply node/node.go:1844]: They are ignored. The reason I rename the database file is that the migration isn't executed more than once and instead of deleting I rename it so it could be restored if necessary.

The goal is not that the old db is still usable under a different name.

File: node/node.go (hunk 1 above)
Comment (fasmat) [reply node/node.go:1838]: Theoretically yes, but `VACUUM` does the same thing, with the added benefit of cleaning up / shrinking the DB in the process.

**Review by poszu — APPROVED** [review]

**Comment by fasmat** [comment]
bors merge

**Comment by fasmat** [comment]
bors merge

**Comment by fasmat** [comment]
Test flake

bors merge

**Comment by fasmat** [comment]
bors merge

**Comment by fasmat** [comment]
bors merge

**Comment by fasmat** [comment]
test flake

bors merge

**Comment by fasmat** [comment]
bors merge

**Comment by dshulyak** [comment]
i think this is the bug, as actual keys for lookup are not added and preserve own atx is noop https://github.com/spacemeshos/go-spacemesh/blob/develop/node/node.go#L432 

from what i understood this test will work if you don't have to rely on preserve own atx to capture last available atx. in case if last atxs was already delievered to a node that will be used to generate snapshot. hence flakiness the potential flakiness that you implied.

**Comment by fasmat** [comment]
@dshulyak yes the code on `develop` is clearly wrong and although the test passes the nodes complain that ATXs that they should have do not exist:

https://grafana.spacemesh.dev/explore?panes=%7B%22ayE%22:%7B%22datasource%22:%22loki-gke-dev%22,%22queries%22:%5B%7B%22refId%22:%22A%22,%22expr%22:%22%7Bnamespace%3D%5C%22test-vqmi%5C%22%7D%20%7C%3D%20%60ERROR%60%22,%22queryType%22:%22range%22,%22datasource%22:%7B%22type%22:%22loki%22,%22uid%22:%22loki-gke-dev%22%7D,%22editorMode%22:%22builder%22%7D%5D,%22range%22:%7B%22from%22:%221709690306721%22,%22to%22:%221709694002721%22%7D%7D%7D&schemaVersion=1&orgId=1

That's why it failed initially on my PR because it failed to compute eligibilities due to missing ATXs. I fixed the code that preserves and restores own ATXs of a node and now although all nodes (eventually) have all ATXs they need I see different issues and the test never completes (because none of the nodes create proposals after the restore):

https://grafana.spacemesh.dev/explore?panes=%7B%22ayE%22:%7B%22datasource%22:%22loki-gke-dev%22,%22queries%22:%5B%7B%22refId%22:%22A%22,%22expr%22:%22%7Bnamespace%3D%5C%22test-mpbe%5C%22%7D%20%7C%3D%20%60ERROR%60%20%21%3D%20%60failed%20to%20compute%20eligibilities%60%22,%22queryType%22:%22range%22,%22datasource%22:%7B%22type%22:%22loki%22,%22uid%22:%22loki-gke-dev%22%7D,%22editorMode%22:%22builder%22%7D%5D,%22range%22:%7B%22from%22:%22now-12h%22,%22to%22:%22now%22%7D%7D%7D&schemaVersion=1&orgId=1

**Comment by dshulyak** [comment]
yes sorry i didn't notice that you fixed the same problem i pointed out, and then stumbled on another one

**Comment by fasmat** [comment]
bors merge
